### PR TITLE
Change special folder success to loading and change navigation

### LINKED ISSUE #7553: [Account setup] Don't allow navigating back to transient special folders screen
When special folders are automatically configured by the server, the user only sees transient screens ("Fetching list of folders…" and "All special folders have been configured automatically by the server"). Currently we allow the user to change the special folder assignment anyway. That's also why we allow the user to navigate back to this screen. 

image

We've now decided that we want to keep things simple and don't allow users to change the special folders assignment (during account setup) when the server provided a complete mapping. In that case, the success screen is completely transient and we don't need the "Edit special folders" or "Next" buttons. When navigating back from the following screen, the special folders screen should be skipped.

However, if the server didn't provide a complete mapping, we prompt the user to configure special folders. In that case, navigating back from the screen that follows, should display that special folders screen again.

### PR DESCRIPTION
**Pull request by wmontwe**
Don't allow navigating back to transient special folders screen

Fixes #7553

### CODE REVIEW

File: feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/navigation/AccountSetupNavHost.kt [hunk 1]
```diff
@@ -130,7 +130,17 @@ fun AccountSetupNavHost(
         composable(route = NESTED_NAVIGATION_SPECIAL_FOLDERS) {
             SpecialFoldersScreen(
                 onNext = {
-                    navController.navigate(NESTED_NAVIGATION_DISPLAY_OPTIONS)
+                    navController.navigate(NESTED_NAVIGATION_DISPLAY_OPTIONS) {
+                        if (hasSpecialFolders) {
```
Comment (cketti) [inline feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/navigation/AccountSetupNavHost.kt:134]: `hasSpecialFolders` is `true` when the special folders screen should be displayed. It doesn't indicate whether special folders have been configured automatically. (Maybe we need to change the name)

My suggestion is to add an additional callback parameter `onAutoAdvance` to `SpecialFoldersScreen` to be called when special folders have been found and the screen advanced automatically. 

That way we can pop `SpecialFoldersScreen` off the back stack when auto advancing (`onAutoAdvance`) and leave it on the back stack when the user had to configure special folders and manually pressed "Next" (`onNext`).

**Review by cketti — CHANGES_REQUESTED** [review]
To be consistent with the other transient screens the "Next" button should only be shown when the user has to manually specify the folder mapping.

File: feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersViewModel.kt [hunk 2]
```diff
@@ -129,23 +129,14 @@ class SpecialFoldersViewModel(
 
     private fun navigateNext() {
```
Comment (cketti) [inline feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersViewModel.kt:130]: By changing this to `fun navigateNext(isManualSetup: Boolean)` we can pass in the appropriate value at the call sites and don't need `State.isManualSetup`.

File: feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersScreenKtTest.kt [hunk 3]
```diff
@@ -33,7 +33,7 @@ class SpecialFoldersScreenKtTest : ComposeTest() {
         assertThat(onNextCounter).isEqualTo(0)
         assertThat(onBackCounter).isEqualTo(0)
 
-        viewModel.effect(Effect.NavigateNext)
+        viewModel.effect(Effect.NavigateNext(true))
```
Comment (cketti) [inline feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersScreenKtTest.kt:36]: Please use a named argument here and in all the other `Effect.NavigateNext` constructor calls in tests.

**Review by cketti — APPROVED** [review]

File: feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersScreenKtTest.kt (hunk 3 above)
Comment (wmontwe) [reply feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersScreenKtTest.kt:36]: You always have fresh ideas, why not writing them down for everyone to participate, thanks.

Comment (cketti) [reply feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/specialfolders/SpecialFoldersScreenKtTest.kt:36]: Some things are hard to write down in a formal rule. But it always comes down to "how easy is this code to read?".

Easy to read: `enableProgressIndicator(true)`
Somewhat verbose, but fine: `enableProgressIndicator(enable = true)`
Really verbose: `enableProgressIndicator(enableProgressIndicator = true)`

Leaves the reader wondering about the parameter: `Effect.NavigateNext(true)`
The reader has a better idea what the argument is about: `Effect.NavigateNext(isManualSetup = true)`

**Pull request merged**